components/mpas-ocean/src/shared/mpas_ocn_vel_self_attraction_loading.F

```diff
@@ -0,0 +1,2444 @@
+                Slm(:)=(0.0_dp, 0.0_dp)
+
+                ! Rearrange ssh into CellID order
+                do iCell = 1,nCellsGlobal
+                   globalArray(indexToCellIDGathered(iCell)) = gatheredArray(iCell)
+                enddo
+
+                ! Interpolate ssh onto Gaussian Grid
+                call interpolate( toColValues, toRowValues, toSValues, globalArray, ssh_gg)
+                ! Perform spherical harmonic transform
+#ifdef USE_SHTNS
+                call Spat_to_SH(shtns_c, ssh_gg, Slm)
+                call mpas_timer_stop('Serial SAL: Forward')
+                call mpas_timer_start('Serial SAL: Inverse')
+                ! Multiply each harmonic coefficient by the scaling factor
+                do ilm = 1,shtns%nlm
+                    Slm(ilm) = Slm(ilm) * LoveScaling(ilm)
+                enddo
+                ! Perform inverse spherical harmonic transform
+                call SH_to_spat(shtns_c, Slm, ssh_gg)
+#endif
+                ! Interpolate back to MPAS mesh
+                call interpolate( fromColValues, fromRowValues, fromSValues, ssh_gg, globalArray)
+
+                ! Rearrange back to index order
+                do iCell = 1,nCellsGlobal
+                    gatheredArray(iCell) = globalArray(indexToCellIDGathered(iCell))
+                enddo
+                call mpas_timer_stop('Serial SAL: Inverse')
+            else
+                call mpas_timer_start('Serial SAL: Forward')
+                call mpas_timer_stop('Serial SAL: Forward')
+                call mpas_timer_start('Serial SAL: Inverse')
+                call mpas_timer_stop('Serial SAL: Inverse')
+            endif
+
+            ! Scatter back to ssh_sal
+    !!mrp check nCellsAll on next line
+            call mpas_timer_start('Serial SAL: Scatter')
+            call MPI_SCATTERV(gatheredArray, nCellsPerProc, nCellsDisplacement, MPI_DOUBLE, &
+                        ssh_sal, nCellsAll, MPI_DOUBLE, 0, dminfo % comm, err)
+
+            if (curProc.eq.0) then
+              deallocate(globalArray, gatheredArray, ssh_gg)
+            endif
+
+            ! Perform Halo exchange update
+            call mpas_dmpar_field_halo_exch(domain,'ssh_sal')
+            call mpas_timer_stop('Serial SAL: Scatter')
+
+        else
+
+            !$omp parallel
+
+            call mpas_timer_start('Parallel SAL: Forward Transform')
+            !!!!!!!!!!!!!!!!!!!!!
+            ! Forward Transform
+            !!!!!!!!!!!!!!!!!!!!!
+
+            !$omp do schedule(runtime) private(l,n)
+            do m = 0,nOrder
+                do n = m,nOrder
+                    l = SHOrderDegreeToIndex(n,m)
+                    SnmRe_local(l) = 0.0_RKIND
+                    SnmIm_local(l) = 0.0_RKIND
+                    SnmRe(l) = 0.0_RKIND
+                    SnmIm(l) = 0.0_RKIND
+                enddo
+            enddo
+            !$omp end do
+
+            !$omp do schedule(runtime) private(blk,startIdx,endIdx,n,l,iCell)
```

Comment:
@sbrus89 I updated this. Please double check. I added all indices interior to the m loop.

Reply:
@mark-petersen, I saw this. Those should be private by default, but it is better to be explicit.